crates/bevy_ecs/hecs/src/archetype.rs

```diff
@@ -290,13 +257,17 @@ impl Archetype {
         }
     }
 
-    fn grow(&mut self, increment: u32) {
+    fn grow(&mut self, increment: usize) {
         unsafe {
-            let old_count = self.len as usize;
-            let count = old_count + increment as usize;
-            let mut new_entities = vec![!0; count].into_boxed_slice();
-            new_entities[0..old_count].copy_from_slice(&self.entities[0..old_count]);
-            self.entities = new_entities;
+            let old_count = self.len;
+            let count = old_count + increment;
+            self.entities.resize(
+                self.entities.len() + increment,
+                Entity {
+                    id: !0,
+                    generation: !0,
```

Comment:
```suggestion
                    id: u32::MAX,
                    generation: u32::MAX,
```
I tend to prefer these for clarity. Not an important nit though.

Reply:
100% agreed